presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryStreamReader.java

```diff
@@ -243,4 +255,16 @@ public String toString()
                 .addValue(streamDescriptor)
                 .toString();
     }
+
+    @Override
+    public void close()
+    {
+        systemMemoryContext.close();
```

Comment:
**NOTE:** Please make this change (null the vectors on close) to other stream readers.

We can null the vectors. You don't need to change `getRetainedSizeInBytes()` as `sizeOf` handles nulls.
```
nullVector = null;
dataVector = null;
inDictionaryVector = null;
```